addons/binding/org.openhab.binding.hp1000/src/main/java/org/openhab/binding/hp1000/handler/HP1000Handler.java

```diff
@@ -0,0 +1,116 @@
+        super(thing);
+        this.channelTypeRegistry = channelTypeRegistry;
+    }
+
+    private State buildChannelState(Channel channel, String value) {
+        String acceptedItemType = channel.getAcceptedItemType();
+        if (acceptedItemType == null) {
+            return UnDefType.NULL;
+        }
+        switch (acceptedItemType) {
+            case "Number":
+                Double doubleValue = StateConverterUtils.parseDouble(value);
+                if (doubleValue != null) {
+                    doubleValue = formatUnits(channel, doubleValue);
+                }
+                return StateConverterUtils.toDecimalType(doubleValue);
+            case "String":
+                return StateConverterUtils.toStringType(value);
+            case "DateTime":
+                return StateConverterUtils.toDateTimeType(value);
+            default:
+                logger.error("accepted item type {} not handeled", channel.getAcceptedItemType());
+        }
+        return UnDefType.NULL;
+    }
+
+    private Double formatUnits(Channel channel, Double value) {
+        ChannelType channelType = channelTypeRegistry.getChannelType(channel.getChannelTypeUID());
+        if (channelType == null) {
+            return value;
+        }
+        String pattern = channelType.getState().getPattern();
+        if (pattern.contains("°C")) {
+            return UnitConverterUtils.fahrenheitToCelius(value);
+        } else if (pattern.contains("km/h")) {
+            return UnitConverterUtils.mphTokmh(value);
+        } else if (pattern.contains("mm")) {
+            return UnitConverterUtils.inchesToMillimeters(value);
+        } else if (pattern.contains("hPa")) {
+            return UnitConverterUtils.inhgToHpa(value);
+        }
+
+        return value;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // not supported
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    public void webHookEvent(Map<String, String[]> paramterMap) {
+        getThing().getChannels().stream().forEach(channel -> {
+            ChannelTypeUID channelTypeUID = channel.getChannelTypeUID();
+            if (channelTypeUID == null) {
+                return;
+            }
+            String channelId = channelTypeUID.getId();
+            Optional<String> parameterKey = paramterMap.keySet().stream().filter(key -> key.equalsIgnoreCase(channelId))
+                    .findFirst();
+            if (parameterKey.isPresent()) {
+                String[] parameterValues = paramterMap.get(parameterKey.get());
+                if (parameterValues.length > 0) {
+                    updateState(channel.getUID(), buildChannelState(channel, parameterValues[0]));
+                }
+            } else {
+                logger.error("parameter map not contains channelId {}", channelId);
```

Comment:
Is that really an error? I think if the map does not contain this parameter you can just skip that channel, can't you?